### ISSUE TITLE
Sending SMS (not allowed!)

### ISSUE THREAD

**Issue by SourcemakerHD** [WhisperSystems/TextSecure#2245]
I configured TextSecure to not use the SMS feature. So no SMS messages should be sent. Since the last update TextSecure sents SMS messages which are not allowed!

**Comment by chmac** [comment]
I also have this issue. I have all outgoing SMS blocked by  I was trying to send messages to a friend. He unregistered from the PUSH service, and registered with a new number. I went into the old conversation, and I replied to a message. That message was sent by SMS (at significant cost, I'm roaming!). First type of unallowed SMS send.

Second type, I was messaging on the PUSH service, then lost WiFi momentarily, and again, TextSecure sent an SMS without my permission.

In my case, TextSecure is not my default messaging app. I also have some kind of permission manager running, which I've just used to block TextSecure's "Send SMS" permission. Hopefully that avoids this happening again for me, but it appears that this is definitely an issue which can affect other users.

**Comment by rbu** [comment]
There's several possible bugs in this issue.

@SourcemakerHD mentioned in the third comment, part (1) that it may be possible to start a conversation with a non-TS user (where no conversation exists yet). How can you reproduce this? I could not reproduce.

@SourcemakerHD mentioned in (2) that an existing conversation (via PUSH) turns into an SMS conversation when the other person unregisters PUSH. This is a problem I could reproduce and which I would like to raise as well. Sending a large message, a friend of mine encountered the cost of a dozen SMS when someone else unregistered.
When SMS is set to "Nobody" this needs to be respected even in existing conversations!

@chmac raised a third issue which is the app falling back to SMS when PUSH is temporarily unavailable. I cannot reproduce. Are you sure the SMS was actually sent? Or do you have a green error message?

**Comment by chmac** [comment]
In the case where PUSH was temporarily unavailable, the UI shows me the same green message, and when I press and hold on it, then click the info (i), it tells me it was sent via SMS. However, @rbu you're absolutely right, it didn't actually send the message. It also shows me a red triangle on the left, which I guess means "message failed to send".

**Comment by chmac** [comment]
As a point of feedback, showing the same green message for failed messages or for messages sent via the SMS transport is a little confusing. I realise (now) that a small red triangle has been added, but that wasn't clear to me immediately that the message had not been sent. Might be a nice idea to make the background red or grey instead of the green, to show that the message had not been sent at all.

**Comment by SourcemakerHD** [comment]
@rbu
There was already a conversation, because a friends wrote my an SMS, before installing TS.

**Comment by rbu** [comment]
@chmac, agreed (regarding the color). Could you open a separate bug report for this please?

@SourcemakerHD, ok. Then this sounds like the problem I have observed as well. I would like to emphasize this issue is rather critical, as users are billed for SMS sending when they explicitly disabled it.

**Comment by chmac** [comment]
@rbu Done #2308 

Agreed, the issue of SMS messages being sent when they're explicitly disabled feels critical to me, I spent money without knowing it (while roaming), albeit not very much.

**Comment by SourcemakerHD** [comment]
The problem still exists with the latest update :-1:

**Comment by agrajaghh** [comment]
I guess #2350 is going to fix that

**Comment by forestvogo** [comment]
Hi, just using the latest version I think 2.4.2 on Sony Z1 compact. Moved to a new phone, but SIM was not working, when I set up the phone. Exported all messages on the old phone and imported on the new phone. Changed the settings for TextSecure to no SMS in/outgoing and no MMS in/outgoing.

Tried to set up a new secure message chat with a friend and send about 4 messages, which where all not delivered. Registering the new phone was impossible, as the SIM was not working yet.

Today retrieved a new SIM with working network. Before registering the new SIM, TextSecure send the 4 SMS to my friend, while  SMS/MMS in/outgoing was disabled.

Push messages was unchecked do to not working SIM and no reregistering possibilities.

Guess this refers to the same bug. Further questions?
